tests/cpgrid/lgr/LgrChecks.hpp

```diff
@@ -161,18 +169,23 @@ void checkGlobalActiveCellsCountInGridWithLgrs(const Dune::CpGrid& grid,
 } // namespace Opm
 
 void Opm::checkReferenceElemParentCellVolume(Dune::cpgrid::HierarchicIterator it,
-                                             const Dune::cpgrid::HierarchicIterator& endIt)
+                                             const Dune::cpgrid::HierarchicIterator& endIt,
+                                             bool isNested)
 {
     double reference_elem_parent_cell_volume = std::accumulate(it, endIt, 0.0,
                                                                [](double sum, const Dune::cpgrid::Entity<0>& child) {
                                                                    return sum + child.geometryInFather().volume();
                                                                });
-    BOOST_CHECK_CLOSE(reference_elem_parent_cell_volume, 1.0, 1e-12);
+    if (!isNested) {
```

Comment:
I think we should do this differently. 

I guess the problem was that the hierarchic iterator iterates over all levels until the specified level. Hence we need to either make sure that only iteratates over one level (by specifying that level as max level) or only accumulate above for one level and skip elements on other levels.